river/utils/math.py

```diff
@@ -29,31 +29,56 @@
     "woodbury_matrix",
 ]
 
+##################################################################################
+###ADD_THIS_FUNCTION_FOR_XSTREAM##################################################
+
+def get_minmax_array(X):
+	"""Utility method that returns the boundaries for each feature of the input array.
+    Args:
+        X (np.float array of shape (num_instances, num_features)): The input array.
+    Returns:
+        min (np.float array of shape (num_features,)): Minimum values for each feature in array.
+        max (np.float array of shape (num_features,)): Maximum values for each feature in array.
+	"""
+	min = np.min(X, axis=0)
+	max = np.max(X, axis=0)
+
+	return min, max
+    
+def _iterate(X, y=None):
+    """Iterates array of features and possibly labels.
+    Args:
+        X (np.array of shape (num_instances, num_features)): The features array.
+        y (np.array of shape (num_instances, ): The array containing labels (Default=None).
+    """
+
+    iterator = ArrayStreamer(shuffle=False)
+
+    if y is None:
+        for xi in iterator.iter(X):
+            yield xi, None
+    else:
+        for xi, yi in iterator.iter(X, y):
+            yield xi, yi
+##################################################################################
 
 def dotvecmat(x, A):
     """Vector times matrix from left side, i.e. transpose(x)A.
-
     Parameters
     ----------
     x
     A
-
```

Comment:
Your code editor is removing empty lines, but it shouldn't be. They increase readibility